api/src/services/security-service.ts

```diff
@@ -76,4 +84,86 @@ export class SecurityService extends DBService {
       )
     );
   }
+
+  /**
+   * Returns the signed URL of a document for which a user has permissions.
+   *
+   * Rules:
+   * non-admin user cannot access the document when:
+   ** - it is pending review, OR
+   ** - user hasn't been granted an exception to every security rule
+   *
+   * non-admin user can access the document when:
+   * - document is not secured
+   * - user has the correct exceptions
+   *
```

Comment:
nit: I like that you added the rules here, I think we should just be more explicit in exactly the conditions that allow someone access or not, and don't hesitate to state the obvious either. 

For example: `document is not secure` could be more like `A security review has been completed (security review is no longer pending) and the document is deemed not secure (no security rules were applied).`

Similar for `user has the correct exceptions`. That is true IF the document has had a review completed and was found to be secure, and had security rules applied.

So just stuff like that, being overly descriptive and exactly how these states come about.

Reply:
I would even potentially mirror some of the JSDoc comments in the code below where they are applicable, just to make it foolproof for future developers who need to modify this code.